src/TraceEvent/Ctf/CtfMetadataLegacyParser.cs

```diff
@@ -39,6 +39,8 @@ public CtfMetadataLegacyParser(Stream stream)
         public override IEnumerable<CtfMetadataDeclaration> Parse()
         {
             string metadata = GetMetadata();
+            // add new line before string in case some metadata doesn't start with \n (kernel metadata)
+            metadata = "\n" + metadata;
```

Comment:
Can you describe why this is needed?

Reply:
In line 60 ``` int start = metadata.Substring(0, open).LastIndexOf('\n') + 1;``` the parser always looks for a new line before parsing a metadata declaration. This doesn't work for kernel metadata, which doesn't start with a new line and the first declaration will not be found by this code. I need to do more tests to know if my assumption that "only kernel metadata begins without a new line'' but it is more safe to add a new line before the metadata string for either case. Please let me know if there's better approach than this hack...